packages/discovery-react-components/src/components/SearchFacets/SearchFacets.tsx

```diff
@@ -49,6 +49,17 @@ interface SearchFacetsProps {
    * Number of facet terms to show when list is collapsed
    */
   collapsedFacetsCount?: number;
+  /**
+   * Used for analytics tracking
+   */
+  trackEventFacetsSelect?: (payload: {
```

Comment:
I don't think we want to expose properties related to tracking. What we can do is expose custom user interactions (like onClick or onKeyUp, etc) then reuse those props to add tracking events in tooling

Reply:
@maniax89 okay, I understand not passing the needed properties up to the tooling track event function that exists there.  But I feel like I'm already doing what you're saying, I'm passing callback functions down into components that trigger the track event function call in tooling.  I'm probably misinterpreting what you're saying, but could you clarify a bit more or perhaps provide some psuedo example?  That would be helpful.

Reply:
its more about naming and reusability more than anything. once we commit to these properties, changing them requires a breaking change. 

we could just have a prop exposed called `onChange` that gets called with the native input events from checkboxes and radio buttons get triggered. that way users have access to all the same properties they would normally when invoking `onChange` events - see how carbon exposes their `onChange` event https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/Checkbox/Checkbox.js#L45
```
onChange?: (e: SyntheticEvent<HTMLInputElement>) => void
```

then if that function is present, call it when triggering change events on the checkbox in https://github.com/watson-developer-cloud/discovery-components/blob/master/packages/discovery-react-components/src/components/SearchFacets/components/FacetsGroups/MultiSelectFacetsGroup.tsx#L67
```typescript
  const handleOnChange = (
    checked: boolean,
    _id: string,
    event: SyntheticEvent<HTMLInputElement>
  ): void => {
    if (onChange) {
      onChange(event);
    }
   // rest of handling code
  };
```

you will have to change some of the currently internal `onChange` symbols (they shouldn't have been named that anyway) to something more custom like `onFacetChange` ex:
```diff
<FieldFacets
            allFacets={allFieldFacets}
            showMatchingResults={showMatchingResults}
            onChange={onChange}
+.          onFacetChange={handleOnChange}
            collapsedFacetsCount={collapsedFacetsCount}
            messages={mergedMessages}
          />
```

Reply:
I really like this idea ^^^. It's clean and generic (not just tied to analytics).

Reply:
@maniax89 I've updated the PR with changes mostly to `SearchFacets`.  Wanted to know if this is the right direction I should be going in before attempting to clean things up.

I'm also in a situation where in the `SearchInput` component, I'm not entirely sure on how to proceed with that one.  It seems like there are two different onChange events that could be surfaced.  The `onChange` for input entry, and an `onKeyUp` for pressing the enter key to submit.  Each on their own are not that useful for segment tracking, but together, some logic could be cobbled together to make a useful track event.  Though I'm sure if we even want to do that.  Also those `onChange` events don't really give enough indication on whether the search query was typed in manually or selected using the autocomplete.